ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -488,6 +496,35 @@ public static long toSecondsPrecisionInMs(long ms) {
   }
 
   /**
+   * Gets the {@link GetOptions} required by the request.
+   * @param restRequest the representation of the request.
+   * @return the required {@link GetOptions}. Defaults to {@link GetOptions#None}.
+   * @throws RestServiceException if the {@link RestUtils.Headers#GET_OPTIONS} is present but not recognized.
+   */
+  public static GetOptions getGetOptions(RestRequest restRequest)
+      throws RestServiceException {
+    GetOptions options = GetOptions.None;
```

Comment:
+1. Can change once @cgtz takes a complete look since I don't want those changes messing up the diff.